src/Workspaces/Core/Portable/Utilities/AsyncLazy`1.cs

```diff
@@ -294,6 +294,13 @@ private Request CreateNewRequest_NoLock()
 
         public override Task<T> GetValueAsync(CancellationToken cancellationToken)
         {
+            var cachedResult = _cachedResult;
+            if (cachedResult != null)
+            {
+                // Fastest possible path after the computation completes
+                return cachedResult;
+            }
```

Comment:
This behavior differs from `GetValue`.  In that method the `cancellationToken` value takes precedence over whether or not the task is already completed.  This considers a completed result before `cancellationToken`.  Seems odd.